src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { getVisibleTodos } from './getVisibleTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [status, setStatus] = useState('full');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    try {
+      getTodos()
+        .then(setTodos);
+    } catch (error) {
+      setTodos([]);
```

Comment:
What does this catch do? Empty array already in the todo state.

Reply:
**Fixed**